src/https_client.c

```diff
@@ -282,20 +285,33 @@ static void https_fetch_ctx_init(https_client_t *client,
 }
 
 static int https_fetch_ctx_process_response(https_client_t *client,
-                                            struct https_fetch_ctx *ctx)
+                                            struct https_fetch_ctx *ctx,
+                                            int curl_result_code)
 {
   CURLcode res = 0;
   long long_resp = 0;
   char *str_resp = NULL;
   int faulty_response = 1;
 
+  switch (curl_result_code) {
+    case CURLE_OK:
+      DLOG_REQ("curl request succeeded");
+      faulty_response = 0;
+      break;
+    case CURLE_WRITE_ERROR:
+      ELOG_REQ("Response content was too large");
```

Comment:
Is this guaranteed? Maybe a bit more direct about this? 
e.g. "curl write error (response too large?)"

Reply:
It is guaranteed. Search for CURLE_WRITE_ERROR in the following 2 pages of curl docs: [doc1](https://curl.se/libcurl/c/libcurl-errors.html) [doc2](https://curl.se/libcurl/c/CURLOPT_WRITEFUNCTION.html)
But of course, I can change it.